esp-hal/src/soc/esp32s3/psram.rs

```diff
@@ -168,16 +168,17 @@ pub(crate) fn init_psram(config: PsramConfig) {
         const MMU_INVALID: u32 = 1 << 14;
         const DR_REG_MMU_TABLE: u32 = 0x600C5000;
 
-        // calculate the PSRAM start address to map
-        let mut start = EXTMEM_ORIGIN;
+        // calculate the PSRAM start address to map - honor the fact there might be
+        // unmapped pages in between
```

Comment:
To be quite honest, since I don't know anything about this problem, this comment doesn't tell me anything. Where are unmapped pages (i.e. what does "in between" mean here), and why are they a problem? I need to read the old code and the diff together to make sense of this and given that we don't have tests, I think we need some better explanations around this.

Reply:
I tried to explain things a bit better - happy to get suggestions to explain it better

Reply:
I simply don't understand this part of the system enough. I don't understand why we're counting flash pages or what the relationship is between ICACHE and PSRAM, when we seem to be mapping PSRAM for the data bus, which is backed by DCACHE.

I also don't know where anything related is documented, a pointer to that would also be helpful.

Reply:
mhhh there is not too much information about the MMU unfortunately - hope the link helps a bit